operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -408,16 +408,17 @@ private void processIntrospectionResult(Packet packet, String result) {
       }
 
       private NextAction handleFailure(Packet packet, V1Job domainIntrospectorJob) {
+        Step nextSteps = null;
         Optional.ofNullable(domainIntrospectorJob).ifPresent(job -> logIntrospectorFailure(packet, job));
 
-        Step nextSteps = Step.chain(
-              createFailureRelatedSteps(Introspection, onSeparateLines(severeStatuses)),
-              getNextStep(packet, domainIntrospectorJob));
-
         if (!severeStatuses.isEmpty()) {
           nextSteps = Step.chain(createFailureCountStep(domainIntrospectorJob), nextSteps);
         }
 
+        nextSteps = Step.chain(
+              createFailureRelatedSteps(Introspection, onSeparateLines(severeStatuses)),
+              getNextStep(packet, domainIntrospectorJob), nextSteps);
```

Comment:
I believe getNextSteop() method can return null. When it is null, will we have a null step in the middle of the chain here?

Reply:
My understanding was that Step.chain can handle null elements in the middle of the chain. The reason I made the above change is because we want the `FailureCountStep` to be executed last (i.e. after the `FailedStep`) to retain the correct message in domain status. If `FailedStep` is executed last, it overwrites the domain status message created by the `FailureCount` step. I didn't see any issues in the unit or the integration test with this approach. Please let me know if my understanding is incorrect, I can try to come up with some other approach.

Reply:
From the Javadoc of `Step.chain`, the parameter stepGroups is multiple groups of steps. Any ideas if the group of steps have to be non-null?
```
  /**
   * Chain the specified step groups into a single chain of steps.
   *
   * @param stepGroups multiple groups of steps
   * @return the first step of the resultant chain
   */
```

Reply:
Look at StepChainTest.ignoreNullMiddleSteps. It actually verifies that this case works - null steps are ignored.

Reply:
Thanks, Russ. We are good then.

Reply:
Ideally, the FailureCount step should be after the FailedStep and before createEventStep so that the event can be populated with the domain status message. This will help to match the event message with the domain status message. If this requires too much changes, we can consider to make this change in a future PR.

Reply:
Changed in ea2d103.